payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CountryConfig.kt

```diff
@@ -13,27 +13,70 @@ import java.util.Locale
  * @property onlyShowCountryCodes: a list of country code that should be shown.  If empty all
  * countries will be shown.
  * @property locale: this is the locale used to display the country names.
+ * @property flagMode: when true, will display items in "flag mode", a smaller form which shows the
+ * flags of the countries before their name in the dropdown list, and only the flag when the list is
+ * collapsed.
  */
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 class CountryConfig(
     val onlyShowCountryCodes: Set<String> = emptySet(),
-    val locale: Locale = Locale.getDefault()
+    val locale: Locale = Locale.getDefault(),
+    val flagMode: Boolean = false
 ) : DropdownConfig {
     override val debugLabel = "country"
 
     @StringRes
     override val label = R.string.address_label_country
 
-    override fun getDisplayItems(): List<String> = CountryUtils.getOrderedCountries(locale)
+    override val tinyMode = flagMode
+
+    internal val countries = CountryUtils.getOrderedCountries(locale)
         .filter {
-            onlyShowCountryCodes.isEmpty() ||
-                (onlyShowCountryCodes.isNotEmpty() && onlyShowCountryCodes.contains(it.code.value))
-        }.map { it.name }
+            onlyShowCountryCodes.isEmpty() || onlyShowCountryCodes.contains(it.code.value)
+        }
+
+    override val displayItems: List<String> = countries.map { country ->
+        if (flagMode) {
+            "${countryCodeToEmoji(country.code.value)} ${country.name}"
```

Comment:
I like the idea of always showing the flag on the country dropdown, not just enabled through the flagMode.

Reply:
Done